### PR TITLE
Update eslint config

### PR DESCRIPTION
**Pull request by brmscheiner**
@Atticus29 here is the output of running eslint for me with the package.json from this PR:

```
ben@BScheiner-MBP16 codex-frontend %  NODE_ENV=development BABEL_ENV=development ./node_modules/.bin/eslint --fix src/**/*.{js,jsx}
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating

/Volumes/dev/codex-frontend/src/components/AnnotatedPhotograph.jsx
  28:5  error  Visible, non-interactive elements with click handlers must have at least one keyboard listener  jsx-a11y/click-events-have-key-events
  28:5  error  Static HTML elements with event handlers require a role                                         jsx-a11y/no-static-element-interactions

/Volumes/dev/codex-frontend/src/components/AnnotationCreator.jsx
  186:6  error  React Hook useCallback has missing dependencies: 'imgSrc', 'theme.palette.common.white', and 'theme.palette.primary.main'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/components/AnnotationEditor.jsx
  111:6  error  React Hook useCallback has missing dependencies: 'annotations', 'imgSrc', 'theme.palette.primary.main', 'theme.palette.secondary.dark', and 'theme.palette.secondary.main'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/components/IndividualSelector.jsx
  48:27  error  Declare this component outside parent component "IndividualSelector" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components
  59:27  error  Declare this component outside parent component "IndividualSelector" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components

/Volumes/dev/codex-frontend/src/components/Link.jsx
  38:7  error  Using target="_blank" without rel="noreferrer" (which implies rel="noopener") is a security risk in older browsers: see https://mathiasbynens.github.io/rel-noopener/#recommendations  react/jsx-no-target-blank

/Volumes/dev/codex-frontend/src/components/ProfileUploader.jsx
  50:6  error  React Hook useEffect has missing dependencies: 'assetSubmissionId' and 'onComplete'. Either include them or remove the dependency array. If 'onComplete' changes too often, find the parent component that defines it and wrap that definition in useCallback  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/components/SadScreen.jsx
  40:5  error  React Hook useMemo has missing dependencies: 'rest' and 'variantOverrides'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/components/UserProfile.jsx
  56:5  error  React Hook useMemo has a missing dependency: 'someoneElse'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/components/cards/CollaborationsCard.jsx
   98:27  error  Declare this component outside parent component "CollaborationsCard" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components
  117:27  error  Declare this component outside parent component "CollaborationsCard" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components

/Volumes/dev/codex-frontend/src/components/cards/CooccurrenceCard.jsx
  21:33  error  Declare this component outside parent component "CooccurrenceCard" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components

/Volumes/dev/codex-frontend/src/components/cards/EncounterCard.jsx
  31:5  error  React Hook useMemo has an unnecessary dependency: 'siteSettingsVersion'. Either exclude it or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/components/cards/EncountersCard.jsx
  52:5   error  React Hook useMemo has missing dependencies: 'encounters', 'intl', and 'regionOptions'. Either include them or remove the dependency array                         react-hooks/exhaustive-deps
  52:6   error  React Hook useMemo has a complex expression in the dependency array. Extract it to a separate variable so it can be statically checked                             react-hooks/exhaustive-deps
  92:27  error  Declare this component outside parent component "EncountersCard" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components

/Volumes/dev/codex-frontend/src/components/cards/GalleryCard.jsx
  23:22  error  The href attribute is required for an anchor to be keyboard accessible. Provide a valid, navigable address as the href value. If you cannot provide an href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid

/Volumes/dev/codex-frontend/src/components/cards/IndividualCard.jsx
  27:5   error  React Hook useEffect has a missing dependency: 'setIndividualData'. Either include it or remove the dependency array. If 'setIndividualData' changes too often, find the parent component that defines it and wrap that definition in useCallback  react-hooks/exhaustive-deps
  71:13  error  Fragments should contain more than one child - otherwise, there‘s no need for a Fragment at all                                                                                                                                                    react/jsx-no-useless-fragment

/Volumes/dev/codex-frontend/src/components/cards/RelationshipsCard.jsx
  187:5  error  React Hook useMemo has a missing dependency: 'individualGuid'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/components/cards/SightingsCard.jsx
  51:5   error  React Hook useMemo has missing dependencies: 'intl', 'regionOptions', and 'sightings'. Either include them or remove the dependency array                         react-hooks/exhaustive-deps
  51:6   error  React Hook useMemo has a complex expression in the dependency array. Extract it to a separate variable so it can be statically checked                            react-hooks/exhaustive-deps
  89:27  error  Declare this component outside parent component "SightingsCard" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components

/Volumes/dev/codex-frontend/src/components/dataDisplays/IndividualsDisplay.jsx
   27:27  error  Declare this component outside parent component "IndividualsDisplay" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components
  133:7   error  Expected no line gap between “showNoResultsBao” and “rest”                                                                                                             react/jsx-props-no-multi-spaces

/Volumes/dev/codex-frontend/src/components/dataDisplays/Paginator.jsx
  15:27  error  Unsafe arithmetic operation on optional chaining. It can result in NaN  no-unsafe-optional-chaining
  15:50  error  Unsafe arithmetic operation on optional chaining. It can result in NaN  no-unsafe-optional-chaining
  17:39  error  Unsafe arithmetic operation on optional chaining. It can result in NaN  no-unsafe-optional-chaining
  22:15  error  Unsafe arithmetic operation on optional chaining. It can result in NaN  no-unsafe-optional-chaining

/Volumes/dev/codex-frontend/src/components/dataDisplays/RelationshipDisplay.jsx
  24:27  error  Declare this component outside parent component "RelationshipDisplay" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components

/Volumes/dev/codex-frontend/src/components/dataDisplays/SightingsDisplay.jsx
  59:27  error  Declare this component outside parent component "SightingsDisplay" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components
  83:27  error  Declare this component outside parent component "SightingsDisplay" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components

/Volumes/dev/codex-frontend/src/components/dataDisplays/cellRenderers/OverflowController.jsx
  37:7  error  Arrow function expected no return value  consistent-return

/Volumes/dev/codex-frontend/src/components/fields/edit/LatLongEditor.jsx
  46:7   error  React Hook useMemo does nothing when called with only one argument. Did you forget to pass an array of dependencies?                      react-hooks/exhaustive-deps
  68:5   error  React Hook useEffect has a missing dependency: 'savedMapLatLng'. Either include it or remove the dependency array                         react-hooks/exhaustive-deps
  68:6   error  React Hook useEffect has a complex expression in the dependency array. Extract it to a separate variable so it can be statically checked  react-hooks/exhaustive-deps
  68:32  error  React Hook useEffect has a complex expression in the dependency array. Extract it to a separate variable so it can be statically checked  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/components/fields/edit/LocationIdEditor.jsx
  30:5  error  React Hook useMemo has a missing dependency: 'schema'. Either include it or remove the dependency array                                 react-hooks/exhaustive-deps
  30:6  error  React Hook useMemo has a complex expression in the dependency array. Extract it to a separate variable so it can be statically checked  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/components/fields/view/LocationIdViewer.jsx
  36:5   error  React Hook useMemo has missing dependencies: 'choices' and 'intl'. Either include them or remove the dependency array                   react-hooks/exhaustive-deps
  36:13  error  React Hook useMemo has a complex expression in the dependency array. Extract it to a separate variable so it can be statically checked  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/components/filterFields/mapUtils/PointDistanceMap.jsx
  77:5  error  React Hook useEffect has a missing dependency: 'draw'. Either include it or remove the dependency array                                 react-hooks/exhaustive-deps
  86:5  error  React Hook useEffect has missing dependencies: 'draw', 'latitude', and 'longitude'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/components/inputs/FileInput.jsx
  77:6  error  React Hook useEffect has missing dependencies: 'onChange' and 'schema.allowedFileTypes'. Either include them or remove the dependency array. If 'onChange' changes too often, find the parent component that defines it and wrap that definition in useCallback  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/components/maps/Marker.jsx
  20:5  error  React Hook useEffect has a missing dependency: 'entry'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/components/report/Cropper.jsx
  67:6  error  React Hook useCallback has missing dependencies: 'imgSrc', 'theme.palette.primary.main', 'theme.palette.secondary.dark', and 'theme.palette.secondary.main'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/components/report/UploadManager.jsx
  74:6  error  React Hook useEffect has missing dependencies: 'assetSubmissionId', 'onUploadComplete', 'onUploadStarted', and 'setFiles'. Either include them or remove the dependency array. If 'onUploadComplete' changes too often, find the parent component that defines it and wrap that definition in useCallback  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/components/settings/SettingsFileUpload.jsx
  69:6  error  React Hook useEffect has missing dependencies: 'allowedFileTypes', 'assetSubmissionId', 'maxFileSize', 'onSetPostData', and 'settingName'. Either include them or remove the dependency array. If 'onSetPostData' changes too often, find the parent component that defines it and wrap that definition in useCallback  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/hooks/useDimensions.js
  28:6  error  React Hook useEffect has missing dependencies: 'ref' and 'setDimensions'. Either include them or remove the dependency array. If 'setWidth' needs the current value of 'ref', you can also switch to useReducer instead of useState and read 'ref' in the reducer  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/hooks/useDocumentTitle.js
  37:5  error  React Hook useEffect has missing dependencies: 'intl', 'messageValues', and 'translateMessage'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/hooks/useMutate.js
   78:9   error  iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations  no-restricted-syntax
   79:11  error  Unexpected `await` inside a loop                                                                                                                                           no-await-in-loop
  104:5   error  React Hook useEffect has a missing dependency: 'success'. Either include it or remove the dependency array                                                                 react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/hooks/useOptions.js
  17:19  error  React Hook "useMemo" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?  react-hooks/rules-of-hooks
  51:5   error  React Hook useMemo has a missing dependency: 'data'. Either include it or remove the dependency array                                                                                      react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/models/encounter/useEncounterFieldSchemas.js
   24:33  error  React Hook "useMemo" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?  react-hooks/rules-of-hooks
  107:5   error  React Hook useMemo has missing dependencies: 'data' and 'intl'. Either include them or remove the dependency array                                                                         react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/models/identification/useIdConfigSchemas.js
  87:5  error  React Hook useMemo has a missing dependency: 'data'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/models/individual/useFilterIndividuals.js
  64:22  error  Missing radix parameter  radix

/Volumes/dev/codex-frontend/src/models/individual/useIndividualFieldSchemas.js
  70:5  error  React Hook useMemo has a missing dependency: 'data'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/models/server/useServerStatus.js
  134:6  error  React Hook useEffect has a missing dependency: 'initialState'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/models/sighting/useFilterSightings.js
  38:22  error  Missing radix parameter  radix

/Volumes/dev/codex-frontend/src/models/sighting/useSightingFieldSchemas.js
  156:5  error  React Hook useMemo has missing dependencies: 'data', 'detectionConfig', and 'intl'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/models/users/useUserMetadataSchemas.js
  89:5  error  React Hook useMemo has a missing dependency: 'includeEmail'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/assetGroup/AGSTable.jsx
  51:27  error  Declare this component outside parent component "AGSTable" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components

/Volumes/dev/codex-frontend/src/pages/bulkImport/BulkReportForm.jsx
  115:5  error  React Hook useEffect has missing dependencies: 'numEncounterFieldsForFlatFile' and 'numSightingFieldsForFlatFile'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/bulkImport/utils/useBulkImportFields.js
   88:5  error  React Hook useMemo has a missing dependency: 'intl'. Either include it or remove the dependency array                        react-hooks/exhaustive-deps
  126:5  error  React Hook useMemo has missing dependencies: 'intl' and 'regionOptions'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/createIndividual/CreateIndividual.jsx
  50:5  error  React Hook useEffect has a missing dependency: 'createFieldSchemas'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/fieldManagement/settings/CategoryTable.jsx
   85:27  error  Declare this component outside parent component "FieldSettings" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components
  104:27  error  Declare this component outside parent component "FieldSettings" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components

/Volumes/dev/codex-frontend/src/pages/fieldManagement/settings/CustomFieldTable.jsx
   80:29  error  Declare this component outside parent component "CustomFieldTable" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components
   93:29  error  Declare this component outside parent component "CustomFieldTable" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components
  118:5   error  React Hook useMemo has missing dependencies: 'fieldTypeName' and 'intl'. Either include them or remove the dependency array                                          react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/fieldManagement/settings/DefaultFieldTable.jsx
  69:5   error  React Hook useEffect has a missing dependency: 'siteSettings'. Either include it or remove the dependency array. If 'setFormSettings' needs the current value of 'siteSettings', you can also switch to useReducer instead of useState and read 'siteSettings' in the reducer  react-hooks/exhaustive-deps
  77:27  error  Declare this component outside parent component "DefaultFieldTable" or memoize it. If you want to allow component creation in props, set allowAsProps option to true                                                                                                           react/no-unstable-nested-components
  91:27  error  Declare this component outside parent component "DefaultFieldTable" or memoize it. If you want to allow component creation in props, set allowAsProps option to true                                                                                                           react/no-unstable-nested-components

/Volumes/dev/codex-frontend/src/pages/fieldManagement/settings/FieldDemo.jsx
  31:5  error  React Hook useEffect has a missing dependency: 'demoFieldValue'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/fieldManagement/settings/FileTypeEditor.jsx
  70:22  error  Do not use Array index in keys  react/no-array-index-key

/Volumes/dev/codex-frontend/src/pages/fieldManagement/settings/defaultFieldComponents/SpeciesEditor.jsx
  51:27  error  Declare this component outside parent component "SpeciesEditor" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components
  85:27  error  Declare this component outside parent component "SpeciesEditor" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components

/Volumes/dev/codex-frontend/src/pages/fieldManagement/settings/saveField/SaveField.jsx
  100:5  error  React Hook useEffect has missing dependencies: 'id', 'newField', and 'type'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/generalSettings/GeneralSettings.jsx
  122:6  error  React Hook useEffect was passed a dependency list that is not an array literal. This means we can't statically verify whether you've passed the correct dependencies  react-hooks/exhaustive-deps
  122:6  error  React Hook useEffect has a missing dependency: 'edmValues'. Either include it or remove the dependency array                                                          react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/generalSettings/IntelligentAgentSettings.jsx
  45:5  error  React Hook useEffect has missing dependencies: 'intelligentAgentSettingsFields' and 'setIntelligentAgentFieldsValid'. Either include them or remove the dependency array. Outer scope values like 'intelligentAgentSchema' aren't valid dependencies because mutating them doesn't re-render the component  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/individual/Individual.jsx
   81:24  error  Unsafe arithmetic operation on optional chaining. It can result in NaN                                       no-unsafe-optional-chaining
   98:5   error  React Hook useMemo has a missing dependency: 'intl'. Either include it or remove the dependency array        react-hooks/exhaustive-deps
  142:44  error  Unsafe arithmetic operation on optional chaining. It can result in NaN                                       no-unsafe-optional-chaining
  157:5   error  React Hook useMemo has a missing dependency: 'encounters'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
  169:5   error  React Hook useMemo has a missing dependency: 'intl'. Either include it or remove the dependency array        react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/individual/components/PictureBookGallery.jsx
   16:9  error  Return values from promise executor functions cannot be read                                                                                                                                                                 no-promise-executor-return
  134:5  error  React Hook useEffect has a missing dependency: 'onLoad'. Either include it or remove the dependency array. If 'onLoad' changes too often, find the parent component that defines it and wrap that definition in useCallback  react-hooks/exhaustive-deps
  144:5  error  React Hook useEffect has a missing dependency: 'onLoad'. Either include it or remove the dependency array. If 'onLoad' changes too often, find the parent component that defines it and wrap that definition in useCallback  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/match/ImageCard.jsx
  28:5  error  React Hook useMemo has an unnecessary dependency: 'siteSettingsVersion'. Either exclude it or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/match/MatchSighting.jsx
  158:5  error  React Hook useEffect has a missing dependency: 'selectedQueryAnnotation'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/reportSighting/ReportForm.jsx
   90:5  error  React Hook useMemo has an unnecessary dependency: 'siteSettingsVersion'. Either exclude it or remove the dependency array                                                                                      react-hooks/exhaustive-deps
  149:5  error  React Hook useEffect has missing dependencies: 'customEncounterSchemas', 'customSightingSchemas', 'defaultEncounterSchemas', and 'defaultSightingSchemas'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/settings/Settings.jsx
  50:5  error  React Hook useEffect has a missing dependency: 'formValues'. Either include it or remove the dependency array. You can also do a functional update 'setFormValues(f => ...)' if you only need 'formValues' in the 'setFormValues' call  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/setup/SiteSetupForm.jsx
  47:6  error  React Hook useEffect was passed a dependency list that is not an array literal. This means we can't statically verify whether you've passed the correct dependencies  react-hooks/exhaustive-deps
  47:6  error  React Hook useEffect has a missing dependency: 'edmValues'. Either include it or remove the dependency array                                                          react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/sighting/AnnotationDetail.jsx
  41:5  error  React Hook useMemo has a missing dependency: 'annotation'. Either include it or remove the dependency array                             react-hooks/exhaustive-deps
  41:6  error  React Hook useMemo has a complex expression in the dependency array. Extract it to a separate variable so it can be statically checked  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/sighting/AssetGroupSighting.jsx
  27:5  error  React Hook useEffect has missing dependencies: 'hash' and 'history'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/sighting/EditSightingMetadata.jsx
  73:5  error  React Hook useEffect has a missing dependency: 'metadata'. Either include it or remove the dependency array                               react-hooks/exhaustive-deps
  73:6  error  React Hook useEffect has a complex expression in the dependency array. Extract it to a separate variable so it can be statically checked  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/sighting/SightingHistoryDialog.jsx
  174:25  error  The href attribute is required for an anchor to be keyboard accessible. Provide a valid, navigable address as the href value. If you cannot provide an href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid

/Volumes/dev/codex-frontend/src/pages/sighting/encounters/EditEncounterMetadata.jsx
  76:5  error  React Hook useEffect has a missing dependency: 'metadata'. Either include it or remove the dependency array                               react-hooks/exhaustive-deps
  76:6  error  React Hook useEffect has a complex expression in the dependency array. Extract it to a separate variable so it can be statically checked  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/sighting/identification/CustomMatchingSetForm.jsx
  39:5  error  React Hook useEffect has missing dependencies: 'algorithms', 'regionSchema', and 'setIdConfig'. Either include them or remove the dependency array. If 'setIdConfig' changes too often, find the parent component that defines it and wrap that definition in useCallback  react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/splashSettings/SplashSettings.jsx
  74:6  error  React Hook useEffect was passed a dependency list that is not an array literal. This means we can't statically verify whether you've passed the correct dependencies  react-hooks/exhaustive-deps
  74:6  error  React Hook useEffect has a missing dependency: 'edmValues'. Either include it or remove the dependency array                                                          react-hooks/exhaustive-deps

/Volumes/dev/codex-frontend/src/pages/userManagement/UserEditTable.jsx
  36:27  error  Declare this component outside parent component "UserEditTable" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components
  46:27  error  Declare this component outside parent component "UserEditTable" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components
  67:27  error  Declare this component outside parent component "UserEditTable" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components

/Volumes/dev/codex-frontend/src/pages/userManagement/UserManagerCollaborationEditTable.jsx
  122:27  error  Declare this component outside parent component "UserManagersCollaborationEditTable" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components
  132:27  error  Declare this component outside parent component "UserManagersCollaborationEditTable" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components
  144:27  error  Declare this component outside parent component "UserManagersCollaborationEditTable" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components
  154:27  error  Declare this component outside parent component "UserManagersCollaborationEditTable" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components
  167:27  error  Declare this component outside parent component "UserManagersCollaborationEditTable" or memoize it. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components

/Volumes/dev/codex-frontend/src/utils/notificationUtils.js
  7:16  error  React Hook "useIntl" is called in function "getNotificationProps" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. React Hook names must start with the word "use"  react-hooks/rules-of-hooks

✖ 121 problems (121 errors, 0 warnings)
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,16 +57,16 @@
...
-    "eslint-plugin-import": "^2.25.3",
-    "eslint-plugin-jsx-a11y": "^6.5.1",
+    "eslint-plugin-import": "~2.25.3",
+    "eslint-plugin-jsx-a11y": "~6.5.1",
     "eslint-plugin-node": "^9.1.0",
     "eslint-plugin-prettier": "^3.0.0",
-    "eslint-plugin-react": "^7.27.0",
-    "eslint-plugin-react-hooks": "^4.3.0",
+    "eslint-plugin-react": "~7.28.0",
+    "eslint-plugin-react-hooks": "~4.3.0",
```
Comment (Emily-Ke) [inline package.json:69]: How did you decide which packages to update and the versions to use?
Also, why not update `eslint` to 8.18.0?

**Review by Atticus29 — APPROVED** [review]
All of this LGTM

File: package.json (hunk 1 above)
Comment (Atticus29) [reply package.json:69]: I almost asked the same question, but I suppose it doesn't really matter for eslint, right?

Comment (brmscheiner) [reply package.json:69]: ahhh, I wanted to continue using eslint-config-airbnb and found this in their documentation:
```
Install the correct versions of each package, which are listed by the command:
npm info "eslint-config-airbnb@latest" peerDependencies
```
so these were the versions provided to me by that command.

https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb/README.md

Comment (Emily-Ke) [reply package.json:69]: When I run that, I get
```
{
  eslint: '^7.32.0 || ^8.2.0',
  'eslint-plugin-import': '^2.25.3',
  'eslint-plugin-jsx-a11y': '^6.5.1',
  'eslint-plugin-react': '^7.28.0',
  'eslint-plugin-react-hooks': '^4.3.0'
}
```
which doesn't explain the downgrade of `eslint-config-prettier` or why you chose `eslint`  8.3.0

Comment (brmscheiner) [reply package.json:69]: It told me eslint 8.3.0, not 8.2.0

**Pull request merged**